packages/nvidia_nat_weave/src/nat/plugins/weave/fastapi_plugin_worker.py

```
@@ -0,0 +1,128 @@
+            async def add_chat_feedback(request: Request, payload: dict):
+                """Add reaction feedback for an assistant message via observability trace ID."""
+
+                async with session_manager.session(http_connection=request,
+                                                   user_authentication_callback=self._http_flow_handler.authenticate):
+                    # Extract parameters from payload
+                    observability_trace_id = payload.get('observability_trace_id')
+                    reaction_type = payload.get('reaction_type')
+
+                    if not observability_trace_id or not reaction_type:
+                        raise HTTPException(status_code=400,
+                                            detail="observability_trace_id and reaction_type are required")
```

Ruff (0.14.8)

Abstract `raise` to an inner function

(TRY301)